storage.go

```diff
@@ -474,6 +474,9 @@ func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairSt
 		Bucket: aws.String(s.name),
 		Key:    aws.String(rp),
 	}
+	if opt.Size > 0 {
```

Comment:
Please use `opt.HasXxxx` to check `size` and `offset`.

It's OK to eliminate the check for `offset`, but please fill comments here for the reason.

Reply:
Modified, Please Review。